homie-controller/src/lib.rs

```diff
@@ -134,6 +134,35 @@ struct PublishResponse {
     topics_to_unsubscribe: Vec<String>,
 }
 
+fn build_node_topics(topics: &mut Vec<String>, base_topic: &str, device_id: &str, node_id: &str) {
```

Comment:
Nit: as this is adding topics to an existing `Vec`, I'd rather it were called something like `add_node_topics`. (And likewise for `build_property_topics` below.)